blades/combo_blade.h

```diff
@@ -0,0 +1,200 @@
+they use different data lines for control.
+
+The resulting blade will have a length of the number of LEDs in the first and
+second blade combined. When the style is running, the second blade's LEDs will
+be controlled as though they were in line and after the first blade's LEDs
+
+For a simple example, if you have a saber with 2 crystal LEDs running on different
+data lines rather than in line with one another, you might often do the following.
+
+{ 0,
+  WS281XBladePtr<128, bladePin, Color8::GRB, PowerPINS<bladePowerPin2, bladePowerPin3> >(),
+  WS281XBladePtr<1, blade4Pin, Color8::GRB, PowerPINS<bladePowerPin2, bladePowerPin3> >(),
+  WS281XBladePtr<1, blade3Pin, Color8::GRB, PowerPINS<bladePowerPin2, bladePowerPin3> >(),
+}
+
+The above way allows us to assign either the same style or different styles to
+both LEDs
+
+However, if we assign the same style the LEDs will do the same thing since they
+are both seen as the first LED in a 1 LED longht blade. To avoid this, we can
+use ComboBlade, which will result in the style being run against an intermediate
+blade with a length of 2, rainbow and stripe styles to move between the two
+
+{ 0,
+  WS281XBladePtr<128, bladePin, Color8::GRB, PowerPINS<bladePowerPin2, bladePowerPin3> >(),
+  ComboBlade(WS281XBladePtr<1, blade4Pin, Color8::GRB, PowerPINS<bladePowerPin2, bladePowerPin3> >(),
+             WS281XBladePtr<1, blade3Pin, Color8::GRB, PowerPINS<bladePowerPin2, bladePowerPin3>>()),
+
+In the above example we must reduce the NUM_BLADES by 1 since the ComboBlade itself
+is the only thing that counts as an entry in the blade array.
+
+*/
+
+class ComboBladeWrapper;
+
+
+/*
+ This class is used by ComboBladeWrapper to allow intercepting the style run()
+ calls sent from the blades inside the ComboBladeWrapper.
+
+ The style of the wrapped blades will be set to an instance of this class, which
+ will then call the real style with our ComboBladeWrapper instance as the
+ active blade.
+
+ Since the same style is being executed for both blades we only actually run
+ the style calls sent by the primary blade.
+
+*/
+class ComboBladeStyleWrapper : public BladeStyle {
+public:
+  // Bladestyle implementation
+  virtual void activate() override {
+    real_style_->activate();
+  }
+  virtual void deactivate() override {
+    real_style_->deactivate();
+  }
+  virtual void run(BladeBase* blade) override;
+
+  bool NoOnOff() override {
+    return real_style_->NoOnOff();
+  }
+
+  virtual bool Charging() { return real_style_->Charging();}
+
+  bool IsHandled(HandledFeature effect) override {
+    if (real_style_)
+      return real_style_->IsHandled(effect);
+    return false;
+  }
+
+  OverDriveColor getColor(int i) override {return real_style_->getColor(i);}
+
+  int get_max_arg(int arg) override { return real_style_->get_max_arg(arg);}
+
+  void setRealStyle(BladeStyle* style){real_style_ = style;}
+
+  BladeStyle* GetRealStyle(){return real_style_;}
```

Comment:
needs 3 more more spaces, should be:
  BladeStyle* GetRealStyle() { return real_style_; }

(same comment applies in other places as well.)